addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -124,28 +159,91 @@ options.registry.SocialMedia = options.Class.extend({
     /**
      * @override
      */
-    _computeWidgetState: function (methodName, params) {
+    async _computeWidgetState(methodName, params) {
         if (methodName !== 'renderListItems') {
             return this._super(methodName, params);
         }
-        const listEntries = [];
-        for (const anchorEl of this.$target[0].querySelectorAll(':scope > a')) {
-            const dbField = anchorEl.href.split('/website/social/')[1];
-            const entry = {
-                id: anchorEl.id,
-                selected: !anchorEl.classList.contains('d-none'),
-                display_name: dbField ? this.dbSocialValues['social_' + dbField] : anchorEl.getAttribute('href'),
-                undeletable: Boolean(dbField),
-                placeholder: `https://${dbField || 'example'}.com/yourPage`,
+        await this._fetchSocialMedia();
+        let listPosition = 0;
+        let domPosition = 0;
+        // Check the DOM to compute the state of the ListUserValueWidget.
+        let entries = [...this.$target[0].querySelectorAll(':scope > a')].map(el => {
+            const media = el.href.split('/website/social/')[1];
+            // Avoid a DOM entry and a non-dom entry having the same position.
+            while (this.entriesNotInDom.find(entry => entry.listPosition === listPosition)) {
+                listPosition++;
+            }
+            return {
+                id: generateHTMLId(),
```

Comment:
Probably useless?

Reply:
I've investigated it a little bit, if we do without this ID, then errors occur when two elements have the same value in their we-list input. In fact the problem already exists before this PR, the we-list don't handle the case when two elements of the list have the same value in the input. Drop a form, put the same name to two checkboxes, toggle one of them, both are toogle, even worse, now that both are toggled, impossible to deselect one of them as long as they have the same value in their fields. So actually, for the other bocks that use the we-list, it didn't really make sense to put the same value for two items in the list. But as the social media block is bound to have sometimes several links with the same value (typically the empty string ''), I think it's better to leave this ID for now (and maybe fix the we-list entirely in a future task). What do you think?